doc/api18970.md

```diff
@@ -0,0 +1,334 @@
+# Draft material to review in support of [#18970][]
+
+[//]: # (cmark-gfm doc/api18970.md > /tmp/api18970.html)
+
+[#18970]: https://github.com/zephyrproject-rtos/zephyr/issues/18970
+
+## Background
+
+This section summarizes existing and relevant Zephyr concepts,
+introducing clarifications where necessary.  See also the Zephyr
+[glossary][].
+
+[glossary]: https://docs.zephyrproject.org/latest/glossary.html
+
+### Thread Terminology
+
+Thread behavior in Zephyr is documented in three locations which in
+aggregate are not entirely consistent:
+* [Threads](https://docs.zephyrproject.org/latest/reference/kernel/threads/index.html)
+* [Scheduling](https://docs.zephyrproject.org/latest/reference/kernel/scheduling/index.html)
+* The [Interrupt](https://docs.zephyrproject.org/latest/guides/porting/arch.html#interrupt-and-exception-handling)
+  and [Context Switching](https://docs.zephyrproject.org/latest/guides/porting/arch.html#thread-context-switching)
+  sections of the [Architecture Porting Guide](https://docs.zephyrproject.org/latest/guides/porting/arch.html)
+
+Zephyr defines [six thread
+states](https://docs.zephyrproject.org/latest/reference/kernel/threads/index.html#thread-states)
+of which four are active:
+
+* **Ready** when there is nothing that prevents the thread from becoming
+  active as the current thread, but it is not the current thread.
+* **Running** when the thread is active as the current thread (on a
+  processor).
+* **Waiting** when the thread is on a queue waiting for an event to occur
+  that will transition it to *Ready*.
+* **Suspended** when the thread is inactive and must be transitioned to
+  *Ready* explicitly (e.g. via `k_thread_resume()`.
+
+A thread is made **unready** if it transitions to *Suspended*,
+*Waiting*, or *Terminated*.
+
+Zephyr defines [two mechanisms for selecting the running
+thread](https://docs.zephyrproject.org/latest/reference/kernel/scheduling/index.html#scheduling):
+* In **cooperative** scheduling a thread transitions from *Running* only
+  when it explicitly invokes an operation that makes it unready
+  (i.e. *Waiting* or *Suspended*) or invokes `k_yield()`.
```

Comment:
```suggestion
  (i.e. *Suspended*, Waiting*, or *Terminated*), or invokes `k_yield()`.
```

Reply:
Also, actually, is this true?

What if I call `k_thread_abort(some_coop_thread_id)` from another CPU on SMP?

Not super critical to the terminology below, but it seems like that ought to make the aborted thread unready by these definitions.

Reply:
Reworded to acknowledge external changes to thread state.